code/game/objects/structures/safe.dm

```diff
@@ -224,20 +319,60 @@ FLOOR SAFES
 /obj/structure/safe/examine_status(mob/user)
 	return
 
+/obj/structure/safe/Destroy()
+	global_safes -= src
+	QDEL_NULL(soundloop)
+	QDEL_NULL(spark_system)
+	return ..()
+
+/obj/structure/safe/process()
+	if(drill_timer)
+		overlays -= bar
+		bar = image('icons/effects/progessbar.dmi', src, "prog_bar_[round((((world.time - drill_start_time) / time_to_drill) * 100), 5)]", HUD_LAYER)
+		overlays += bar
+		if(prob(15))
+			spark_system.start()
+
+/obj/structure/safe/examine(mob/user)
+	..()
+	to_chat(user, "On the inside of the the door is <b>[combo_to_open]</b>")

```

Comment:
Won't this just give you the combination? Did you mean to have this conditional on the safe being open?

Reply:
Yes I did, wow how did I miss that. Thank you.